aten/src/ATen/native/cpu/FlashAttentionKernel.cpp

```diff
@@ -370,6 +370,231 @@ void cpu_flash_attention(
 
 }
 
+template <typename scalar_t, int64_t qSplitSize, int64_t kvSplitSize>
+void cpu_flash_attention_backward(
+    const at::Tensor& grad_q,
+    const at::Tensor& grad_k,
+    const at::Tensor& grad_v,
+    const at::Tensor& grad_out,
+    const at::Tensor& query,
+    const at::Tensor& key,
+    const at::Tensor& value,
+    const at::Tensor& out,
+    const at::Tensor& logsumexp,
+    const Tensor& cumulative_sequence_length_q,
+    const Tensor& cumulative_sequence_length_k,
+    const int64_t max_seqlen_batch_q,
+    const int64_t max_seqlen_batch_k,
+    double dropout_p,
+    bool is_causal,
+    const at::Tensor& philox_seed,
+    const at::Tensor& philox_offset,
+    c10::optional<double> scale) {
+
+  using Vec = vec::Vectorized<scalar_t>;
+  float scaling_factor =
+      sdp::calculate_scale(query, scale).as_float_unchecked();
+
+  // Sizes
+  // Query (Batch x Q_seq_len  x Num_heads x Dim_per_head)
+  // Key   (Batch x KV_seq_len x Num_heads x Dim_per_head)
+  // Value (Batch x KV_seq_len x Num_heads x Dim_per_head)
+  int64_t batchSize = query.size(0);
+  int64_t qSize = query.size(1);
+  int64_t kvSize = value.size(1);
+  int64_t num_head = query.size(2);
+  int64_t headSize = query.size(3);
+
+  // Strides
+  int64_t qStrideB = query.stride(0);
+  int64_t qStrideM = query.stride(1);
+  int64_t qStrideH = query.stride(2);
+  int64_t kStrideB = key.stride(0);
+  int64_t kStrideN = key.stride(1);
+  int64_t kStrideH = key.stride(2);
+  int64_t vStrideB = value.stride(0);
+  int64_t vStrideN = value.stride(1);
+  int64_t vStrideH = value.stride(2);
+  int64_t oStrideB = out.stride(0);
+  int64_t oStrideM = out.stride(1);
+  int64_t oStrideH = out.stride(2);
+  int64_t lStrideB = logsumexp.stride(0);
+  int64_t lStrideM = logsumexp.stride(1);
+  int64_t lStrideH = logsumexp.stride(2);
```

Comment:
Can we simplify the data ptr access with `TensorAccessor`? We don't have to retrieve all these strides vars and also it is simpler to get the data ptr with the index variables. Not sure about the perf impact though.

Reply:
Good idea, thanks! As the change is relatively big and the perf impact is also needed to verify, is it OK to do this change in another separate PR?

Reply:
Sure.